pylint/checkers/utils.py

```diff
@@ -1273,6 +1330,27 @@ def is_postponed_evaluation_enabled(node: astroid.node_classes.NodeNG) -> bool:
     return "annotations" in module.future_imports
 
 
+def is_class_subscriptable_pep585_with_postponed_evaluation_enabled(
+    value: astroid.node_classes.NodeNG, node: astroid.node_classes.NodeNG
+) -> bool:
+    """Check if class is subscriptable with PEP 585 and
+    postponed evaluation enabled.
+    """
+    if not is_postponed_evaluation_enabled(node):
+        return False
+
+    if not isinstance(
+        node.parent, (astroid.AnnAssign, astroid.Arguments, astroid.FunctionDef)
+    ):
+        return False
+    if value.name in SUBSCRIPTABLE_CLASSES_PEP585:
```

Comment:
Let's make `SUBSCRIPTABLE_CLASSES_PEP585` and `SUBSCRIPTABLE_CLASSES_PEP585` frozensets rather than tuples.